### PR TITLE
Fix: Reduce or Fix Databricks Notebook Freezing

### PR DESCRIPTION
**Pull request by eakmanrq**
Sometimes Databricks freezes up while applying a plan. Removing the rich terminal progress updating removed all freezing while testing locally. There could still be some issues remaining but this at least fixes one of the sources of freezing. Unfortunately this makes the notebook UI even less appealing but in the short term stability is certainly worth it over presentation.

### CODE REVIEW

File: sqlmesh/core/console.py [hunk 1]
```diff
@@ -927,6 +932,52 @@ def _confirm(self, message: str, **kwargs: t.Any) -> bool:
...
+        """Update snapshot progress."""
+        view_name, loaded_batches, total_batches = self.evaluation_batch_progress[snapshot_name]
+        loaded_batches += 1
+        self.evaluation_batch_progress[snapshot_name] = (view_name, loaded_batches, total_batches)
+        finished_loading = loaded_batches == total_batches
+        status = "Loaded" if finished_loading else "Loading"
+        print(f"{status} '{view_name}', Completed Batches: {loaded_batches}/{total_batches}")
+        if finished_loading:
+            total_finished_loading = len(
+                [x for x in self.evaluation_batch_progress.values() if x[1] == x[2]]
```
Comment (tobymao) [inline sqlmesh/core/console.py:954]: don't need the brackets, leave it a gen

File: sqlmesh/core/console.py [hunk 2]
```diff
@@ -927,6 +932,52 @@ def _confirm(self, message: str, **kwargs: t.Any) -> bool:
...
+        """Indicates that a new promotion progress has begun."""
+        self.promotion_status = (0, total_tasks)
+        print(f"Virtually Updating '{environment}'")
+
+    def update_promotion_progress(self, num_tasks: int) -> None:
+        """Update promotion progress."""
+        num_promotions, total_promotions = self.promotion_status
+        num_promotions += num_tasks
+        self.promotion_status = (num_promotions, total_promotions)
+        if num_promotions % 5 == 0:
```
Comment (tobymao) [inline sqlmesh/core/console.py:974]: if you just did this limiter on rich does it still freeze?

File: sqlmesh/core/console.py (hunk 1 above)
Comment (eakmanrq) [reply sqlmesh/core/console.py:954]: Is this an example of what you mean?
```
len(x for x in [(1,2), (1,4), (5,6)])
```

If so I get this:
```
TypeError: object of type 'generator' has no len()
```

Comment (tobymao) [reply sqlmesh/core/console.py:954]: my bad

**Review by tobymao — APPROVED** [review]

**Pull request merged**